Microsoft.Identity.Web/Resource/AadIssuerValidator.cs

```diff
@@ -163,6 +164,10 @@ private static string GetTenantIdFromToken(SecurityToken securityToken)
             {
                 if (jwtSecurityToken.Payload.TryGetValue(ClaimConstants.Tid, out object tenantId))
                     return tenantId as string;
+
+                // Since B2C doesnt have TID as default, try to get it from iss claims
+                if (jwtSecurityToken.Claims.Any(c => c.Type == ClaimConstants.Acr || c.Type == ClaimConstants.Tfp))

```

Comment:
To solve the B2C known issue of `tid` not being a default claims, it can be extracted from `iss`. The uri format is `{domain}/{tid}/v2.0`